### PR TITLE
In SJ e2e test, check only running jobs

### PR DESCRIPTION
**Pull request by janetkuo**
We should filter "running" jobs in SJ e2e test

**Release note**:

``` release-note
NONE
```

cc @erictune @soltysh

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31474)

### CODE REVIEW

File: pkg/controller/job/utils.go [hunk 1]
```diff
@@ -29,3 +29,13 @@ func IsJobFinished(j *batch.Job) bool {
 	}
 	return false
 }
+
+func FilterActive(jobs *batch.JobList) (active []*batch.Job) {
```
Comment (erictune) [inline pkg/controller/job/utils.go]: Prefer to put this function into the e2e test, and not turn the job controller package into a library.

Comment (janetkuo) [inline pkg/controller/job/utils.go]: Done

**Comment by janetkuo** [comment]
PTAL

**Comment by erictune** [comment]
lgtm

**Comment by janetkuo** [comment]
Squashed and applying tag

**Comment by janetkuo** [comment]
Rebased

**Pull request merged**